src/components/DpTabs/DpTabs.vue

```diff
@@ -6,151 +6,135 @@
+    validator: (prop) => ['medium', 'large'].includes(prop)
   },
 
-  props: {
-    activeId: {
-      type: String,
-      required: false,
-      default: null
-    },
-
-    tabSize: {
-      type: String,
-      required: false,
-      default: 'large',
-      validator: (prop) => ['medium', 'large'].includes(prop)
-    },
-
-    /**
-     * Active tab state may be persisted via an Url fragment. Also, ab clicks are pushed
-     * to the browser history to enable state change when browser navigation is used.
-     */
-    useUrlFragment: {
-      type: Boolean,
-      required: false,
-      default: false
-    }
-  },
+  /**
+   * Active tab state may be persisted via a url fragment. Also, all clicks are pushed
+   * to the browser history to enable state change when browser navigation is used.
+   */
+  useUrlFragment: {
+    type: Boolean,
+    required: false,
+    default: false
+  }
+})
 
-  data: () => ({
-    tabs: []
-  }),
+const emit = defineEmits(['change'])
+const slots = useSlots()
 
-  computed: {
-    tabIds () {
-      return this.tabs.map(tab => tab.id)
-    }
-  },
+const activeTabId = ref(null)
 
-  methods: {
-    /**
-     * When users click the back button, the tabs should behave accordingly.
-     */
-    handleHashChange () {
-      const hash = window.location.hash.slice(1)
-
-      if (this.isTabId(hash)) {
-        this.setActiveTab(hash)
-      }
-    },
-
-    handleTabClick (id) {
-      this.$emit('change', id)
-      this.setActiveTab(id)
-
-      if (this.useUrlFragment) {
-        history.pushState(null, null, `#${id}`)
-      }
-    },
-
-    /**
-     * Check if a given string is a valid tab id.
-     * @param id
-     * @return {boolean}
-     */
-    isTabId (id) {
-      return this.tabs.map(tab => tab.id).includes(id)
-    },
-
-    setActiveTab (id) {
-      this.tabs.forEach(tab => {
-        tab.isActive = (tab.id === id)
-      })
-    },
-
-    /**
-     * If a hash specifies a tab, activate it. If a tab is activated
-     * via prop, activate that tab. Fallback to the first tab.
-     */
-    setInitialTab () {
-      const hash = window.location.hash.slice(1)
-
-      if (this.useUrlFragment && this.isTabId(hash)) {
-        this.setActiveTab(hash)
-      } else if (this.activeId) {
-        this.setActiveTab(this.activeId)
-      } else {
-        this.tabs[0].isActive = true
-      }
-    }
-  },
+const tabs = computed(() => {
+  return slots.default()[0].children.filter(child => child.type.name === 'DpTab')
+})
 
-  created () {
-    this.tabs = this.$children
-  },
+/**
+ * When users click the back button, the tabs should behave accordingly.
+ */
+const handleHashChange = () => {
+  const hash = window.location.hash.slice(1)
 
-  mounted () {
-    this.setInitialTab()
+  if (isTabId(hash)) {
+    setActiveTab(hash)
+  }
+}
 
-    if (this.useUrlFragment) {
-      window.addEventListener('hashchange', this.handleHashChange)
-    }
-  },
+const handleTabChange = (id) => {
```

Comment:
A general Style question: Do we wan't to omit/use the braces if possible? Do we wan't to standardize that?

Reply:
That's a good question. On the one hand, I'd say we should agree on a consistent solution, on the other hand I find omitting the braces here more confusing than in other cases (e.g. `.map()`). What do you think?

Reply:
It is a question of "code reading habit". With `const a = (b) => c`, the "visual signifier" for a function definition was the  parentheses surrounding one or more variable names. Now with `const a = b => c` the arrow is the new visual signifier, and the eye must jump back to the left to look up the arg of the function. My gut feel is that `const a = b => c` is less readable.

Reply:
ok. Than we stick with this notation here.
```
a = (x) => foo
```
but I would say, if the arrow function is a callback like in `.map()` we omit them:
```
a.map(x => foo)
```
Did I get it right?

Reply:
At least that's my opinion. :slightly_smiling_face: We could also have a FE vote?